primitives/src/vrf/blsvrf.rs

```diff
@@ -0,0 +1,132 @@
+//! BLS signature based VRF
+
+use super::Vrf;
+use crate::{
+    constants::CS_ID_BLS_VRF_NAIVE,
+    errors::PrimitivesError,
+    hash_to_group::HashToGroup,
+    signatures::{
+        bls::{BLSSignKey, BLSSignature, BLSVerKey},
+        BLSSignatureScheme, SignatureScheme,
+    },
+};
+use ark_ec::bls12::Bls12Parameters;
+use ark_serialize::CanonicalSerialize;
+use ark_std::{
+    marker::PhantomData,
+    rand::{CryptoRng, RngCore},
+    vec::Vec,
+};
+use digest::Digest;
+
+/// BLS VRF scheme.
+/// Optimized for signature size, i.e.: PK in G2 and sig in G1
+pub struct BLSVRFScheme<P: Bls12Parameters> {
+    pairing_friend_curve: PhantomData<P>,
+}
+
+impl<H, P> Vrf<H, P> for BLSVRFScheme<P>
```

Comment:
Just to be curious: do we need any aggregation APIs regarding VRF?

Reply:
Not aggregation, but BLS-VRFs can be batch verified. We can use that APIs for acceleration.

Reply:
Yeah, make sense.